fairlearn/grid_search/binary_protected_attribute/binary_classification.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import copy
+import numpy as np
+
+from . import utilities
+
+
+def _weight_function(y_val, a_val, L, p_ratio, a0_val):
+    if a_val == a0_val:
+        return 2 * y_val - 1 - L * p_ratio
+    else:
+        return 2 * y_val - 1 + L
+
+
+def _generate_weights(y, protected_attribute, L, p_ratio, a0_val):
+    weight_func = np.vectorize(_weight_function)
+    return weight_func(y, protected_attribute, L, p_ratio, a0_val)
+
+
+def sweep_demographic_parity(learner,
```

Comment:
We should probably cite Miro's paper here? https://arxiv.org/abs/1803.02453
After all, this is described in there.